src/main/java/uk/gov/pay/ledger/transaction/service/TransactionService.java

```diff
@@ -152,8 +153,38 @@ public void upsertTransactionFor(EventDigest eventDigest) {
         transactionDao.upsert(transaction);
     }
 
-    public void upsertTransaction(TransactionEntity transaction) {
-        transactionDao.upsert(transaction);
+    public void upsertRefundTransactionWithPaymentInfo(EventDigest refundEventDigest, EventDigest paymentEventDigest) {
```

Comment:
Previously EventDigestHandler had multiple responsibilities (processing payment, refunds payout events) and has been split into individual processors.
This is moving much of the responsibility of RefundEventProcessor to TransactionService which could be moving us back to previous pattern ? Can we call RefundEventProcessor from PaymentEventProcessor and project refunds ?

Happy to discuss further if this doesn't makes sense

Reply:
Not sure this would entirely make sense to do, as we wouldn't be processing a refund event in this case.

Here's a draft PR of how this will be used https://github.com/alphagov/pay-ledger/pull/879

Maybe this logic could live somewhere else that isn't TransactionService, if it's a problem of TransactionService getting too big?